plugins/GSdx/GSTextureCache.cpp

```diff
@@ -1449,19 +1439,21 @@ GSTextureCache::Source* GSTextureCache::CreateSource(const GIFRegTEX0& TEX0, con
 		if (m_paltex && psm.pal > 0)
 		{
 			src->m_texture = m_renderer->m_dev->CreateTexture(tw, th, Get8bitFormat());
-			src->m_palette = m_renderer->m_dev->CreateTexture(256, 1);
+			src->m_should_have_tex_palette = true;
 		}
-		else
+		else {
 			src->m_texture = m_renderer->m_dev->CreateTexture(tw, th);
+
+			if (psm.pal > 0) {
+				uint16 palette_size = psm.pal * sizeof(uint32);
```

Comment:
This parts of code bugs me. Because it is mostly a copy-past of what it is done when you create/attach a palette.

If I'm right this code is for texture that technically need a palette but texture conversion was done on CPU. 
Do you always need a storage for the CPU side ? In this case, it would be more logical to do in the source constructor. Or did I misunderstand something ?

Reply:
I left the code path for the texture that needs clut copy but have texture conversion done by CPU unchanged.
In fact the clut copy is necessary to ensure correct hit when calling `LookupSource`. I don't think caching clut copies would enhance performances as there is no CPU-GPU communication, and the memcpy alone used here is probably as fast as an hash table hit for clut array.
Also IMHO it is better to move the copy of the clut for CPU converted textures in Source constructor, as the deletion of the array for those is done on Source destruction.
This is therefore out of the scope of the new caching system but for sure I look forward to implement it.